HW2/P2/P2.txt

```diff
@@ -0,0 +1,19 @@
+# ## Time logs
+
+Serial uncorrelated: 0.2906229496 seconds
+Fine grained uncorrelated: 8.40734219551 seconds
+Medium grained uncorrelated: 11.883658886 seconds
+Serial correlated: 0.391420125961 seconds
+Fine grained correlated: 8.02646708488 seconds
+Medium grained correlated: 10.7812769413 seconds
+
+# ## Locking explained
+
+Fine grained: one lock per ceil.
+
+Medium grained: the same lock is shared for each group of ceils of indices between k * N and (k+1) * N in the array.
+
+# ## Medium Grained Value analysis
+
+I plot over 4 iterations the minimum and the average value of the execution time for 12 values of N in both cases, correlated and uncorrelated.
+It seems that for the uncorrelated a bigger value, around 10 is more suited, and for the correlated mostly around 5.
```

Comment:
I'd be curious to see what speed differences you see for correlated data with a larger N, eg. N=20.  I noticed a substantial speed gain at this configuration, which I attributed to the fact that almost all of the src/dest values are within a 10-point spread of each other.  So with an N that exceeds the point spread, we only have to acquire and release one lock instead of two.  Not sure if that's correct, but over a lot of iterations that would explain a speed gain under this regime.